lib/Sema/CSGen.cpp

```diff
@@ -2930,6 +2930,11 @@ namespace {
       
       for (unsigned i : indices(E->getComponents())) {
         auto &component = E->getComponents()[i];
+        auto memberLocator = CS.getConstraintLocator(
+            locator, ConstraintLocator::PathElement::getKeyPathComponent(i));
+        auto resultLocator = CS.getConstraintLocator(
+            memberLocator, ConstraintLocator::FunctionResult);
```

Comment:
Nit: `FunctionResult` is not the best element for the result of the property lookup :)

Reply:
@xedin It's the one already used by subscripts, which is why I chose it. But I don't have to do it that way. What would be an appropriate path element that's not currently used for anything on key path components?

Reply:
I think you can create a `ComponentResult` or locator element per component like `KeyPathPropertyResult`